### ISSUE TITLE
`dart_api_dl` feature is not compile

### ISSUE THREAD

**Issue by rogurotus** [dart-sys/dart-sys#48]
## Describe the bug

compilation does not work when using feature "dart_api_dl" if using crate.io.
path error:
```
thread 'main' panicked at 'ERROR: Failed to generate dart_api_dl binding: 
NotExist("/home/human/.cargo/registry/src/github.com-1ecc6299db9ec823/dart-sys-3.1.8/../dart-sdk/include/dart_version.h")',
/home/human/.cargo/registry/src/github.com-1ecc6299db9ec823/dart-sys-3.1.8/build.rs:647:10
```
works if you use git `dart-sys = { git = "https://github.com/dart-sys/dart-sys.git", features = ["dart_api_dl"] }`. 

## To Reproduce**

1. add `dart-sys = { version = "3.1.8", features = ["dart_api_dl"] }`
2. try compile

## Operating System Info

- Device: PC
- OS: ubuntu

## Additional context
build log

```
[2023-01-23T07:57:30Z] ------------------------------
[2023-01-23T07:57:30Z] INFO: starting build script
```

**Comment by rogurotus** [comment]
@gutenfries ping

**Comment by gutenfries** [comment]
hey, thanks for the ping.

I believe this is due to faulty `include = {}` in the dart-sys subcrate Cargo.toml.

ill get a fix asap and make sure it works. 

I'll also add crates.io integration t3sts to CI :)

**Comment by gutenfries** [comment]
@rogurotus try v3.1.9

**Comment by rogurotus** [comment]
```
error: failed to download `dart-sys v3.1.9`

Caused by:
  unable to get packages from source

Caused by:
  failed to parse manifest at `/home/human/.cargo/registry/src/github.com-1ecc6299db9ec823/dart-sys-3.1.9/Cargo.toml`

Caused by:
  no targets specified in the manifest
  either src/lib.rs, src/main.rs, a [lib] section, or [[bin]] section must be present
  ```

**Comment by rogurotus** [comment]
Apparently it is impossible to specify the parent directory. Also, if `/src` is not specified, it will not be in the package

**Comment by gutenfries** [comment]
Linked PR: #50

**Comment by rogurotus** [comment]
@gutenfries, flutter_rust_bridge found very strange build errors (can't find assert.h (system header search breaks)).
as a solution, don't generate binding.rs in build.rs and use static bindings. This will fix android, macos build errors and improve work through ide

**Comment by gutenfries** [comment]
@rogurotus I agree wholeheartedly.

I've been thinking a lot about code quality, IDE compatibility, and maintainability for Dart-sys; frankly, the solution used in the past isn't sustainable.

I will do more thinking and planning on the subject, but I believe that generating the bindings by an external crate, and merging it with the `update-dart-sdk` crate already in Dart-sys is the most maintainable and clean way to move forward.

I believe that removing `build.rs` entirely may be the best way to move forward.

CC: @OptimisticPeach @fzyzcjy

peer opinions are encouraged, as you may have considered things I have not :)

**Comment by OptimisticPeach** [comment]
Originally when I made the crate the intention was to write down what changes I had made to the generated bindings, and occasionally regenerate them and reapply the changes -- I obviously never ended up doing that because it's a lot of work and I lost interest in working with dart projects.

Here's another idea, although I don't know how difficult it'd be to implement given I've never actually used CI before:

- Every time that a change is pushed to the header file, generate a new commit & version in a separate repo where the crate _actually_ lives. This crate essentially just becomes the machinery to automatically do that (somehow).

**Comment by fzyzcjy** [comment]
> I will do more thinking and planning on the subject, but I believe that generating the bindings by an external crate, and merging it with the update-dart-sdk crate already in Dart-sys is the most maintainable and clean way to move forward.

Not very familiar with code details in dart-sys, so here are my two cents: From experience in frb, as long as the binding in different machines are exactly the same (e.g. I remembered seeing long long or some weird types being different on win/mac), it is great to have binding pre-generated. At the same time, the CI https://github.com/dart-sys/dart-sys/issues/55 can be used to ensure it is always up-to-date so we are safe.

**Comment by gutenfries** [comment]
Marking as closed due to #56, if there is no objection within 48hrs
